### PR TITLE
[Optimization-1987][cdc] Refactoring Flink CDC module

### LINKED ISSUE #1987: [Optimization][cdc] Refactoring Flink CDC module
### Search before asking

- [X] I had searched in the [issues](https://github.com/DataLinkDC/dinky/issues?q=is%3Aissue) and found no similar optimization requirement.

### Description

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by aiwenmo**
## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

File: dinky-cdc/pom.xml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    </modules>
+
+    <properties>
+        <maven.compiler.source>8</maven.compiler.source>
+        <maven.compiler.target>8</maven.compiler.target>
+    </properties>
+
+    <profiles>
+        <profile>
+            <id>flink-1.14</id>
```
Comment (leechor) [inline dinky-cdc/pom.xml:26]: could define modules directly.

File: dinky-cdc/dinky-cdc-plus/src/main/java/org/dinky/cdc/SinkBuilderFactory.java [hunk 2]
```diff
@@ -24,35 +24,29 @@
...
- */
 public class SinkBuilderFactory {
 
     private static final Map<String, Supplier<SinkBuilder>> SINK_BUILDER_MAP =
             new HashMap<String, Supplier<SinkBuilder>>() {
+
                 {
+                    put(SQLSinkBuilder.KEY_WORD, () -> new SQLSinkBuilder());
+                    put(SQLCatalogSinkBuilder.KEY_WORD, () -> new SQLCatalogSinkBuilder());
                     put(KafkaSinkBuilder.KEY_WORD, () -> new KafkaSinkBuilder());
```
Comment (leechor) [inline dinky-cdc/dinky-cdc-plus/src/main/java/org/dinky/cdc/SinkBuilderFactory.java:47]: private static final Map<String, Supplier<SinkBuilder>> SINK_BUILDER_MAP = ImmutableMap.of(
            SQLSinkBuilder.KEY_WORD, SQLSinkBuilder::new,
            SQLCatalogSinkBuilder.KEY_WORD, SQLCatalogSinkBuilder::new,
            KafkaSinkBuilder.KEY_WORD, KafkaSinkBuilder::new,
            DorisSinkBuilder.KEY_WORD, DorisSinkBuilder::new,
            DorisExtendSinkBuilder.KEY_WORD, DorisExtendSinkBuilder::new,
            DorisSchemaEvolutionSinkBuilder.KEY_WORD,
            DorisSchemaEvolutionSinkBuilder::new
    );

**Review by leechor — APPROVED** [review]
LGTM

**Review by Zzm0809 — APPROVED** [review]
LGTM

**Pull request merged**